### PR TITLE
Expand option changes and  document added with different examples.

### PR DESCRIPTION
**Pull request by yerramshilpa**
## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [ ] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [ ] Internet Explorer 11
- [ ] Edge
- [x] Chrome
- [ ] Firefox
- [x] Safari

## Checklist:

- [x] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [x] My code passes the automated linter.
- [x] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts [hunk 1]
```diff
@@ -7,36 +7,63 @@ import * as qs from 'qs';
...
-    const hasValue= qs.stringify(this.model, { skipNulls: true});
-    return this.to.expand  ? true : hasValue ? true : false;
+    if (this.to.hasOwnProperty('expand')) {
+      return this.to.expand;
+    } else {
+      const hasValue =
+        this.formControl.value instanceof Object
+          ? qs.stringify(this.formControl.value, { skipNulls: true })
+          : this.formControl.value;
+      return hasValue ? true : false;
```
Comment (ascreven-gsa) [inline libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts:66]: simplify -- double check Prettier linter

File: libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts [hunk 2]
```diff
@@ -7,36 +7,63 @@ import * as qs from 'qs';
...
-          <ng-container #fieldComponent></ng-container>
-        </sds-accordion-item>
-      </sds-accordion>
-    </ng-container>
-    <ng-container *ngSwitchCase="'panel'">
-      <div class="sds-panel" [ngClass]="{'sds-panel--multiple' : field?.fieldGroup?.length }">
-        <div class="sds-panel__header" *ngIf="!to.hideLabel" [attr.aria-hidden]="!to.announceLabel ? undefined : 'true'"> {{to.label}} </div>
-        <div class="sds-panel__body">
-          <ng-container #fieldComponent></ng-container>
+    <ng-container [ngSwitch]="to.group">
```
Comment (ascreven-gsa) [inline libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts:10]: something causing templates to go over panel... not happening on master
Screen Shot 2020-05-20 at 5 04 30 PM

File: libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.service.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/* tslint:disable */
```
Comment (ascreven-gsa) [inline libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.service.ts:1]: Delete file -- unused

File: libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.data.ts [hunk 4]
```diff
@@ -0,0 +1,4673 @@
+/* tslint:disable */
+export let SampleAutocompleteData = [
```
Comment (ascreven-gsa) [inline libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.data.ts:2]: Delete file -- unused

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  ];
+  formlyFieldConfig: FormlyFieldConfig[] = formlyFieldConfig;
+
+ 
+  constructor(public service: AutocompleteSampleDataService) {
+    this.setup();
+  }
+
+  changes(value) {
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: Removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  providers: [AutocompleteSampleDataService]
+})
+export class FiltersConfig implements OnInit {
+  results: any;
+  form = new FormGroup({});
+  model = {};
+  options: FormlyFormOptions = {};
+  public settings = new SDSAutocompletelConfiguration();
+  public autocompleteModel = new SDSSelectedItemModel();
+  private data = SampleAutocompleteData;
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: Removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import { FormGroup } from '@angular/forms';
+import { FormlyFormOptions, FormlyFieldConfig } from '@ngx-formly/core';
+import { BehaviorSubject } from 'rxjs';
+import { AutocompleteSampleDataService } from './services/autocomplete-sample.service';
+import {
+  SDSAutocompletelConfiguration,
+  SDSSelectedItemModel,
+  SelectionMode
+} from '@gsa-sam/components';
+import { SampleAutocompleteData } from './services/autocomplete-sample.data';
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: Removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+})
+export class FiltersConfig implements OnInit {
+  results: any;
+  form = new FormGroup({});
+  model = {};
+  options: FormlyFormOptions = {};
+  public settings = new SDSAutocompletelConfiguration();
+  public autocompleteModel = new SDSSelectedItemModel();
+  private data = SampleAutocompleteData;
+  public filterChange$ = new BehaviorSubject<object>(null);
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts:14]: Updated code accordingly

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    this.settings.debounceTime = 350;
+  }
+
+  head(array) {
+    return array && array.length ? array[0] : undefined;
+  }
+
+  // To display the selected model values
+  public ngOnInit() {
+    this.filterChange$.subscribe(res => (this.results = res));
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: updated

File: libs/documentation/src/lib/components/filters/filters.module.ts [hunk 10]
```diff
@@ -29,7 +31,14 @@ const DEMOS = {
     code: require('!!raw-loader!./demos/optional/filters-optional.component'),
     markup: require('!!raw-loader!./demos/optional/filters-optional.component.html'),
     path: 'libs/documentation/src/lib/components/filters/demos/optional'
-  }
+  },
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/filters.module.ts:34]: Working now.

File: libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts (hunk 1 above)
Comment (yerramshilpa) [reply libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts:66]: Verified

File: libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.service.ts (hunk 3 above)
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.service.ts]: deleted

File: libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.data.ts (hunk 4 above)
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/services/autocomplete-sample.data.ts]: Deleted

File: libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts (hunk 2 above)
Comment (yerramshilpa) [reply libs/packages/sam-formly/src/lib/formly/wrappers/group.wrapper.ts:10]: Don't see any issues
Screen Shot 2020-05-21 at 1 49 21 PM

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormlyFormOptions, FormlyFieldConfig } from '@ngx-formly/core';
+import { BehaviorSubject } from 'rxjs';
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: updated code

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    this.settings.primaryKeyField = 'id';
+    this.settings.primaryTextField = 'name';
+    this.settings.secondaryTextField = 'subtext';
+    this.settings.labelText = 'Autocomplete';
+    this.settings.selectionMode = SelectionMode.MULTIPLE;
+    this.settings.autocompletePlaceHolderText = 'Enter text';
+    this.settings.debounceTime = 350;
+  }
+
+  head(array) {
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: Removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.html [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<h4>Custom text component inside formly</h4>
+<p><code>expand: true, expand: false </code> are used to set the accordion expand property.
+   If expand is not set and if control has value by defauls the accordion is expanded.</p>
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.html]: updated

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+export class FiltersConfig implements OnInit {
+  results: any;
+  form = new FormGroup({});
+  model = {};
+  options: FormlyFormOptions = {};
+  public settings = new SDSAutocompletelConfiguration();
+  public autocompleteModel = new SDSSelectedItemModel();
+  private data = SampleAutocompleteData;
+  public filterChange$ = new BehaviorSubject<object>(null);
+  fields: FormlyFieldConfig[] = [
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+} from '@gsa-sam/components';
+import { SampleAutocompleteData } from './services/autocomplete-sample.data';
+import { formlyFieldConfig } from './fields';
+
+@Component({
+  templateUrl: './filter-config.component.html',
+  providers: [AutocompleteSampleDataService]
+})
+export class FiltersConfig implements OnInit {
+  results: any;
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts:11]: updated code

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+  SDSAutocompletelConfiguration,
+  SDSSelectedItemModel,
+  SelectionMode
+} from '@gsa-sam/components';
+import { SampleAutocompleteData } from './services/autocomplete-sample.data';
+import { formlyFieldConfig } from './fields';
+
+@Component({
+  templateUrl: './filter-config.component.html',
+  providers: [AutocompleteSampleDataService]
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 17]
```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormlyFormOptions, FormlyFieldConfig } from '@ngx-formly/core';
+import { BehaviorSubject } from 'rxjs';
+import { AutocompleteSampleDataService } from './services/autocomplete-sample.service';
+import {
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 18]
```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormlyFormOptions, FormlyFieldConfig } from '@ngx-formly/core';
+import { BehaviorSubject } from 'rxjs';
+import { AutocompleteSampleDataService } from './services/autocomplete-sample.service';
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 19]
```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts:1]: updated

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+  templateUrl: './filter-config.component.html',
+  providers: [AutocompleteSampleDataService]
+})
+export class FiltersConfig implements OnInit {
+  results: any;
+  form = new FormGroup({});
+  model = {};
+  options: FormlyFormOptions = {};
+  public settings = new SDSAutocompletelConfiguration();
+  public autocompleteModel = new SDSSelectedItemModel();
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+@Component({
+  templateUrl: './filter-config.component.html',
+  providers: [AutocompleteSampleDataService]
+})
+export class FiltersConfig implements OnInit {
+  results: any;
+  form = new FormGroup({});
+  model = {};
+  options: FormlyFormOptions = {};
+  public settings = new SDSAutocompletelConfiguration();
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+        configuration: this.settings,
+        model: this.autocompleteModel,
+        modelChange: this.changes
+      }
+    }
+  ];
+  formlyFieldConfig: FormlyFieldConfig[] = formlyFieldConfig;
+
+ 
+  constructor(public service: AutocompleteSampleDataService) {
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: updated code

File: libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+    this.setup();
+  }
+
+  changes(value) {
+    console.log(value);
+  }
+
+  setup() {
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/filter-config.component.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/fields.ts [hunk 24]
```diff
@@ -0,0 +1,225 @@
...
+export let formlyFieldConfig: FormlyFieldConfig[] = [
+  {
+    key: 'keyword',
+    type: 'input',
+    defaultValue: 10,
+    templateOptions: {
+      type: 'text',
+      label: 'Keyword',
+      expand: false,
+      labelClass: 'usa-sr-only',
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/fields.ts:12]: label class if for screen reader

File: libs/documentation/src/lib/components/filters/demos/config/fields.ts [hunk 25]
```diff
@@ -0,0 +1,225 @@
...
+    key: 'dateWrapper',
+    templateOptions: {
+      label: 'Date',
+      group: 'accordion',
+      expand: true
+    },
+    fieldGroup: [
+      {
+        key: 'modifiedDate',
+        type: 'daterangepicker',
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/fields.ts]: Resolved

File: libs/documentation/src/lib/components/filters/demos/config/fields.ts [hunk 26]
```diff
@@ -0,0 +1,225 @@
...
+        }
+      },
+      {
+        key: 'uniqueEntityIdDuns',
+        type: 'input',
+        templateOptions: {
+          tagText: 'DUNS',
+          tagClass: 'sds-tag--info-purple',
+          label: 'Unique Entity ID',
+          placeholder: '',
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/fields.ts]: updated

File: libs/documentation/src/lib/components/filters/demos/config/fields.ts [hunk 27]
```diff
@@ -0,0 +1,225 @@
...
+          min: 13,
+          max: 40,
+          inputType: 'number',
+          inputStyle: 'error'
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/fields.ts]: removed

File: libs/documentation/src/lib/components/filters/demos/config/fields.ts [hunk 28]
```diff
@@ -0,0 +1,225 @@
...
+        }
+      },
+      {
+        key: 'uniqueEntityIdSam',
+        type: 'input',
+        templateOptions: {
+          tagText: 'SAM',
+          label: 'Unique Entity ID',
+          placeholder: '',
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/fields.ts]: removed

File: libs/packages/sam-formly/src/lib/formly/wrappers/form-field.accordian.ts [hunk 29]
```diff
@@ -10,22 +9,32 @@ import * as qs from 'qs';
...
-    const hasValue= qs.stringify(this.model, { skipNulls: true});
-    return this.to.expand  ? true : hasValue ? true : false;
+    if (this.to.hasOwnProperty('expand')) {
+      return this.to.expand;
+    } else {
+      const hasValue =
+        this.formControl.value instanceof Object
+          ? qs.stringify(this.formControl.value, { skipNulls: true })
+          : this.formControl.value;
+      return hasValue ? true : false;
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/sam-formly/src/lib/formly/wrappers/form-field.accordian.ts:37]: verified

File: libs/documentation/src/lib/components/filters/demos/config/fields.ts [hunk 30]
```diff
@@ -0,0 +1,225 @@
...
+      {
+        key: 'popState',
+        type: 'input',
+        templateOptions: {
+          label: 'State'
+        }
+      }
+    ]
+  },
+  {
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/documentation/src/lib/components/filters/demos/config/fields.ts:201]: Updated wrapper

**Pull request merged**